### PR TITLE
Feature/api name suffix

### PR DESCRIPTION
**Pull request by bgong-mdsol**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

The problem:
API generation for C#, Java and Ruby creates a class based on the tag and then adds "Api" after it. 
Ideally though we have a class like  "Study" or "Study{{Suffix}}" not "StudyApi".

I introduced a apiNameSuffix parameter in order to add suffixes to the generated api class/file/document names. Added the option "--api-name-suffix" to allow the user specify api name suffix via openapi-generator command line, the value of apiNameSuffix is "Api" if the option "--api-name-suffix" is not provided or empty. 

Sample of command line:
java -jar openapi-generator-cli.jar generate g java --api-name-suffix=client -i petstore.yaml -o out/petstore

This PR works for Java, Ruby, C# and Python 

@jcarres-mdsol
@bkashyapmd
@jfeltesse-mdsol
@prajon84
@ykitamura-mdsol
@cabbott

### CODE REVIEW

**Comment by jfeltesse-mdsol** [comment]
At a glance, how about making "Api" as the default suffix for backwards compatibility?
Also, you need to update the README (and other places?) to document that new option.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -1693,17 +1706,17 @@ public String toSetter(String name) {
...
      * @param name the name of the Api
-     * @return capitalized Api name ending with "Api"
+     * @return capitalized Api name
      */
     public String toApiName(String name) {
         if (name.length() == 0) {
             return "DefaultApi";
         }
-        return camelize(name) + "Api";
+        return camelize(name + "_" + apiNameSuffix);
```
Comment (prajon84) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: Is not this code is used by Java and C#, then there should be no `_` before `apiNameSuffix` ?

Comment (bgong-mdsol) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: Method will take care all "_" based on languages

**Comment by bgong-mdsol** [comment]
> At a glance, how about making "Api" as the default suffix for backwards compatibility?
> Also, you need to update the README (and other places?) to document that new option.

@jfeltesse-mdsol updated README and making "Api" as the default suffix  https://github.com/OpenAPITools/openapi-generator/pull/3918/commits/eedc03989ac285fc57aa58a94989c5f862f0fd2d

**Comment by wing328** [comment]
Looks good to me. Thanks for the PR. 

The reason why we add "Api" as the default suffix is to avoid naming collisions with model name (e.g. PetApi, Pet (model))

I'll submit another PR to revise the description a little bit so as to make it clearer that the option only applies to Ruby, Python and JaxRS generator.

Also I'll add the new option to maven-plugin as well as gradle-plugin

**Comment by wing328** [comment]
Doc update done via https://github.com/OpenAPITools/openapi-generator/pull/4045

**Pull request merged**